### PR TITLE
retire sparse_mask_helper

### PR DESCRIPTION
**Pull request by nikitaved**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92976
* #92242
* __->__ #91714

cc @alexsamardzic @pearu @cpuhrsch @amjames @bhosmer @ezyang @gchanan

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Code removal :tada:

**Review by amjames — APPROVED** [review]
Looks good, It is fine to ignore the BC breakage as this was a non-public method.

**Review by cpuhrsch — APPROVED** [review]
Less is more!

File: test/forward_backward_compatibility/check_forward_backward_compatibility.py [hunk 1]
```diff
@@ -322,6 +322,7 @@
     ("aten::_upsample_nearest_exact2d_backward", datetime.date(2022, 12, 15)),
     ("aten::_scaled_dot_product_attention", datetime.date(2023, 3, 15)),
     ("aten::_scaled_dot_product_flash_attention", datetime.date(2023, 3, 15)),
+    ("aten::_sparse_mask_helper", datetime.date(2023, 3, 15)),
     ("aten::_fused_sdp_choice", datetime.date(2023, 3, 15)),
```
Comment (nikitaved) [inline test/forward_backward_compatibility/check_forward_backward_compatibility.py:326]: @cpuhrsch , @albanD , it is an OK to get it removed with the upcoming release, right?

**Pull request closed without merging**